### PR TITLE
More accurate elements commitment tx size estimation

### PR DESCRIPTION
**Pull request by instagibbs**
It seems to be missing some overhead; adjust it to be closer to what it really is and adjust tests as required.

Changelog-None

### CODE REVIEW

File: common/initial_commit_tx.h [hunk 1]
```diff
@@ -28,28 +28,32 @@ static inline size_t commit_tx_base_weight(size_t num_untrimmed_htlcs,
...
-	if (option_anchor_outputs)
+	if (option_anchor_outputs) {
 		weight = 1124;
-	else
+        num_outputs = 4;
+	} else {
 		weight = 724;
-
+        num_outputs = 2;
+    }
```
Comment (vincenzopalazzo) [inline common/initial_commit_tx.h:45]: looks like a couple of tabs are missing?

File: common/initial_commit_tx.h [hunk 2]
```diff
@@ -28,28 +28,32 @@ static inline size_t commit_tx_base_weight(size_t num_untrimmed_htlcs,
...
 	/* BOLT #3:
 	 *
 	 *    2. For each committed HTLC, if that output is not trimmed as
 	 *       specified in [Trimmed Outputs](#trimmed-outputs), add 172
 	 *       to `weight`.
 	 */
 	weight += 172 * num_untrimmed_htlcs;
+    num_outputs += num_untrimmed_htlcs;
```
Comment (vincenzopalazzo) [inline common/initial_commit_tx.h:53]: same there

File: tests/test_connection.py [hunk 3]
```diff
@@ -22,6 +22,9 @@
 import unittest
 import websocket
 
+# Big enough to make channels with 10k effective capacity, including Elements channels
+# which have bigger txns
+CHANNEL_SIZE = 50000
```
Comment (vincenzopalazzo) [inline tests/test_connection.py:27]: maybe we can move this in the `tests/utils.py`?

File: tests/test_gossip.py [hunk 4]
```diff
@@ -20,6 +20,7 @@
 import unittest
 import socket
 
+CHANNEL_SIZE = 50000
```
Comment (vincenzopalazzo) [inline tests/test_gossip.py:23]: same there?

**Review by vincenzopalazzo — COMMENTED** [review]
just a couple of comments, but overall lgtm

File: tests/test_gossip.py (hunk 4 above)
Comment (instagibbs) [reply tests/test_gossip.py]: done

File: tests/test_connection.py (hunk 3 above)
Comment (instagibbs) [reply tests/test_connection.py]: done

File: common/initial_commit_tx.h (hunk 1 above)
Comment (instagibbs) [reply common/initial_commit_tx.h]: fixed, thanks vim

File: common/initial_commit_tx.h (hunk 2 above)
Comment (instagibbs) [reply common/initial_commit_tx.h]: fixed, thanks vim

**Review by vincenzopalazzo — APPROVED** [review]
ACK https://github.com/ElementsProject/lightning/pull/5966/commits/813b4ee199e4f3deeffe6b6a7da4d6f7568fca2f

**Comment by rustyrussell** [comment]
Ack https://github.com/ElementsProject/lightning/commit/91608731a018bdd797c7fdb0ef35d368c9719f5a

**Pull request merged**